File: Views/DailyChallenge/DailyChallengeFlowView.swift
import SwiftUI
import FirebaseAnalytics

struct DailyChallengeFlowView: View {
    @EnvironmentObject var appState: AppState
    @StateObject private var dailyChallengeService = DailyChallengeService.shared
    
    var body: some View {
        Group {
            switch currentRoute {
            case .intro(let showConnect):
                DailyChallengeIntroView(showConnectButton: showConnect)
                    .environmentObject(appState)
                    .onAppear {
                        if !showConnect {
                            // Analytics: intro montrée même si connecté
                            AnalyticsService.shared.track(.introShown(screen: "daily_challenge"))
                        }
                    }
                
            case .paywall(let day):
                DailyChallengePaywallView(challengeDay: day)
                    .environmentObject(appState)
                
            case .main:
                DailyChallengeMainView()
                    .environmentObject(appState)
                
            case .error(let message):
                DailyChallengeErrorView(
                    message: message,
                    onRetry: {
                        configureServiceIfNeeded()
                    }
                )
                .environmentObject(appState)
                
            case .loading:
                DailyChallengeLoadingView()
                    .environmentObject(appState)
            }
        }
        .onAppear {
            configureServiceIfNeeded()
        }
    }
    
    // MARK: - Computed Properties
    
    /// Route actuelle selon l'état de l'application
    private var currentRoute: DailyContentRoute {
        return DailyContentRouteCalculator.calculateRoute(
            for: .dailyChallenge,
            hasConnectedPartner: hasConnectedPartner,
            hasSeenIntro: appState.introFlags.dailyChallenge,
            shouldShowPaywall: shouldShowPaywall,
            paywallDay: currentChallengeDay,
            serviceHasError: false, // DailyChallengeService n'a pas de gestion d'erreur pour l'instant
            serviceErrorMessage: nil,
            serviceIsLoading: dailyChallengeService.isLoading && dailyChallengeService.currentChallenge == nil
        )
    }
    
    /// Vérifier si un partenaire est connecté
    private var hasConnectedPartner: Bool {
        guard let user = appState.currentUser,
              let partnerId = user.partnerId else { return false }
        return !partnerId.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty
    }
    
    /// Calculer le jour actuel du défi
    private var currentChallengeDay: Int {
        if let settings = dailyChallengeService.currentSettings {
            return calculateExpectedDay(from: settings)
        }
        return 1 // Défaut
    }
    
    /// Vérifier si on doit afficher le paywall
    private var shouldShowPaywall: Bool {
        let isSubscribed = appState.currentUser?.isSubscribed ?? false
        if isSubscribed {
            return false // Premium = pas de paywall
        }
        
        // Utiliser FreemiumManager pour vérifier l'accès
        guard let freemiumManager = appState.freemiumManager else { return false }
        return !freemiumManager.canAccessDailyChallenge(for: currentChallengeDay)
    }
    
    private func configureServiceIfNeeded() {
        // 🚨 CORRECTION CRITIQUE: Vérifier partenaire ET intro vue
        guard let currentUser = appState.currentUser, 
              let partnerId = currentUser.partnerId,
              !partnerId.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty else {
            print("⏳ DailyChallengeFlowView: En attente connexion partenaire")
            return
        }
        
        // Vérifier que l'intro a été vue
        guard appState.introFlags.dailyChallenge else {
            print("⏳ DailyChallengeFlowView: En attente intro utilisateur")
            return
        }
        
        // 🚀 OPTIMISATION CACHE: Vérifier si les données sont déjà disponibles ET récentes
        if let currentChallenge = dailyChallengeService.currentChallenge {
            // Vérifier si le défi est pour aujourd'hui
            let today = Date()
            let calendar = Calendar.current
            let challengeDate = currentChallenge.scheduledDate
            
            if calendar.isDate(challengeDate, inSameDayAs: today) {
                print("⚡ DailyChallengeFlowView: Défi d'aujourd'hui déjà disponible - Pas de reconfiguration")
                return
            } else {
                let formatter = DateFormatter()
                formatter.dateStyle = .short
                print("🔄 DailyChallengeFlowView: Défi existant mais pas pour aujourd'hui (\(formatter.string(from: challengeDate)) vs \(formatter.string(from: today)))")
            }
        }
        
        // ✅ Gérer l'accès freemium AVANT de configurer le service
        appState.freemiumManager?.handleDailyChallengeAccess(currentChallengeDay: currentChallengeDay) {
            // Accès autorisé - Configurer le service
            print("🔄 DailyChallengeFlowView: Configuration service pour récupérer défi du jour")
            dailyChallengeService.configure(with: appState)
        }
    }
    
    private func calculateExpectedDay(from settings: DailyChallengeSettings) -> Int {
        let calendar = Calendar.current
        let startOfToday = calendar.startOfDay(for: Date())
        let startOfStartDate = calendar.startOfDay(for: settings.startDate)
        
        let daysSinceStart = calendar.dateComponents([.day], from: startOfStartDate, to: startOfToday).day ?? 0
        return max(1, daysSinceStart + 1)
    }
}



// MARK: - DailyChallengePaywallView

struct DailyChallengePaywallView: View {
    @EnvironmentObject var appState: AppState
    let challengeDay: Int
    @State private var showSubscriptionSheet = false
    
    var body: some View {
        NavigationView {
            ZStack {
                // Fond gris clair identique à la page journal
                Color(red: 0.97, green: 0.97, blue: 0.98)
                    .ignoresSafeArea()
                
                VStack(spacing: 0) {
                    // Header avec titre
                    HStack {
                        Spacer()
                        
                        // Titre
                        VStack(spacing: 4) {
                            Text("paywall_page_title_challenges".localized(tableName: "DailyChallenges"))
                                .font(.system(size: 28, weight: .bold))
                                .foregroundColor(.black)
                        }
                        
                        Spacer()
                    }
                    .padding(.horizontal, 20)
                    .padding(.top, 40) // ✅ PLUS D'ESPACE EN HAUT
                    .padding(.bottom, 100) // Espace augmenté entre titre et image
                    
                    // Contenu principal avec carte floutée
                    VStack(spacing: 30) {
                        // Titre principal
                        Text("paywall_challenges_title".localized(tableName: "DailyChallenges"))
                            .font(.system(size: 24, weight: .bold))
                            .foregroundColor(.black)
                            .multilineTextAlignment(.center)
                            .padding(.horizontal, 20)
                        
                        // Carte de défi floutée
                        ZStack {
                            // Carte simulée avec texte simulé
                            VStack(spacing: 0) {
                                // Header de la carte
                                VStack(spacing: 8) {
                                    Text("daily_challenges_title".localized(tableName: "DailyChallenges"))
                                        .font(.system(size: 18, weight: .bold))
                                        .foregroundColor(.white)
                                        .multilineTextAlignment(.center)
                                }
                                .frame(maxWidth: .infinity)
                                .padding(.vertical, 20)
                                .background(
                                    LinearGradient(
                                        gradient: Gradient(colors: [
                                            Color(red: 1.0, green: 0.4, blue: 0.6),
                                            Color(red: 1.0, green: 0.6, blue: 0.8)
                                        ]),
                                        startPoint: .leading,
                                        endPoint: .trailing
                                    )
                                )
                                
                                // Corps de la carte avec texte simulé
                                VStack(spacing: 30) {
                                    Spacer()
                                    
                                    Text("Envoyez-lui un message pour lui dire pourquoi vous êtes reconnaissant de l'avoir dans votre vie aujourd'hui et partagez trois choses spécifiques que vous appréciez chez lui.")
                                        .font(.system(size: 22, weight: .medium))
                                        .foregroundColor(.white)
                                        .multilineTextAlignment(.center)
                                        .lineSpacing(6)
                                        .padding(.horizontal, 30)
                                        .blur(radius: 8) // FLOU APPLIQUÉ AU TEXTE
                                    
                                    Spacer()
                                    
                                    // Logo/Branding en bas
                                    HStack(spacing: 8) {
                                        Image(systemName: "target")
                                            .font(.system(size: 20))
                                            .foregroundColor(.white.opacity(0.9))
                                        
                                        Text("Jour 4")
                                            .font(.system(size: 16, weight: .semibold))
                                            .foregroundColor(.white.opacity(0.9))
                                    }
                                    .blur(radius: 6) // FLOU APPLIQUÉ AU LOGO
                                    .padding(.bottom, 30)
                                }
                                .frame(maxWidth: .infinity, maxHeight: .infinity)
                                .background(
                                    LinearGradient(
                                        gradient: Gradient(colors: [
                                            Color(red: 0.2, green: 0.1, blue: 0.15),
                                            Color(red: 0.4, green: 0.2, blue: 0.3),
                                            Color(red: 0.6, green: 0.3, blue: 0.2)
                                        ]),
                                        startPoint: .top,
                                        endPoint: .bottom
                                    )
                                )
                            }
                            .frame(maxWidth: .infinity)
                            .cornerRadius(20)
                            .shadow(color: .black.opacity(0.3), radius: 10, x: 0, y: 5)
                            .padding(.horizontal, 20)
                            
                            // Overlay flou et cadenas - SOLUTION AMÉLIORÉE
                            VStack(spacing: 0) {
                                Color.clear
                                    .frame(height: 60) // Espace pour le header non flouté
                                
                                ZStack {
                                    // Fond coloré qui imite la carte défis + flou
                                    LinearGradient(
                                        gradient: Gradient(colors: [
                                            Color(red: 0.2, green: 0.1, blue: 0.15).opacity(0.95),
                                            Color(red: 0.4, green: 0.2, blue: 0.3).opacity(0.95),
                                            Color(red: 0.6, green: 0.3, blue: 0.2).opacity(0.95)
                                        ]),
                                        startPoint: .top,
                                        endPoint: .bottom
                                    )
                                    .blur(radius: 15) // VRAI EFFET DE FLOU
                                    
                                    // Overlay Material pour l'effet glassmorphism
                                    Rectangle()
                                        .fill(.ultraThickMaterial)
                                        .opacity(0.8)
                                    
                                    // Icône coeur
                                    Text("💕")
                                        .font(.system(size: 60))
                                        .shadow(color: .black.opacity(0.3), radius: 2, x: 0, y: 1)
                                }
                            }
                            .cornerRadius(20)
                            .padding(.horizontal, 20)
                        }
                        
                        // Sous-titre
                        Text("paywall_challenges_subtitle".localized(tableName: "DailyChallenges"))
                            .font(.system(size: 16))
                            .foregroundColor(.black.opacity(0.7))
                            .multilineTextAlignment(.center)
                            .padding(.horizontal, 30)
                            .lineLimit(nil) // ✅ PERMETTRE AFFICHAGE COMPLET
                            .fixedSize(horizontal: false, vertical: true) // ✅ TAILLE FLEXIBLE
                    }
                    
                    Spacer(minLength: 40) // ✅ PLUS D'ESPACE AVANT LE BOUTON
                    
                    // Bouton principal conditionnel
                    VStack {
                        Button {
                            showSubscriptionSheet = true
                            
                            // Analytics
                            Analytics.logEvent("cta_premium_clicked", parameters: [
                                "source": "daily_challenge_paywall",
                                "challenge_day": challengeDay
                            ])
                        } label: {
                            Text("paywall_continue_button".localized(tableName: "DailyChallenges"))
                                .font(.system(size: 16, weight: .semibold))
                                .foregroundColor(.white)
                                .padding(.horizontal, 24)
                                .frame(height: 56)
                                .background(
                                    RoundedRectangle(cornerRadius: 28)
                                        .fill(Color(hex: "#FD267A"))
                                )
                        }
                        .padding(.bottom, 160) // Espace pour le menu du bas
                    }
                }
            }
        }

        .sheet(isPresented: $showSubscriptionSheet) {
            SubscriptionView()
                .environmentObject(appState)
        }
    }
}

File: Views/DailyChallenge/DailyChallengeFlowView.kt
// DailyChallengeFlow.kt
package com.love2love.ui.dailychallenge

import android.content.Context
import android.os.Bundle
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.*
import androidx.compose.material3.ModalBottomSheet
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.blur
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import com.google.firebase.analytics.FirebaseAnalytics
import java.time.LocalDate
import java.time.ZoneId
import java.time.temporal.ChronoUnit
import kotlin.math.max

// ---------- Couleurs / Thème locaux ----------
private val PinkPrimary = Color(0xFFFD267A)
private val CardHeaderGradient = Brush.horizontalGradient(
    listOf(Color(0xFFFF6699), Color(0xFFFF99CC))
)
private val CardBodyGradient = Brush.verticalGradient(
    listOf(
        Color(0xFF33242B),
        Color(0xFF664055),
        Color(0xFF99523D)
    )
)
private val PaywallBackdrop = Color(0xFFF7F7FA)

// ---------- Modèle d’état / stubs (à relier à ton app) ----------
data class User(
    val partnerId: String? = null,
    val isSubscribed: Boolean = false
)

data class IntroFlags(val dailyChallenge: Boolean = false)

data class AppState(
    val currentUser: User? = null,
    val introFlags: IntroFlags = IntroFlags(),
    val freemiumManager: FreemiumManager? = null
)

interface FreemiumManager {
    fun canAccessDailyChallenge(day: Int): Boolean
    fun handleDailyChallengeAccess(currentChallengeDay: Int, onAccessGranted: () -> Unit)
}

data class DailyChallenge(val scheduledDate: LocalDate)

data class DailyChallengeSettings(
    val startDate: LocalDate,
    val zoneId: ZoneId = ZoneId.systemDefault()
)

class DailyChallengeService {
    var isLoading by mutableStateOf(false)
        private set

    var currentChallenge by mutableStateOf<DailyChallenge?>(null)
        private set

    var currentSettings by mutableStateOf<DailyChallengeSettings?>(null)
        private set

    fun configure(appState: AppState) {
        // Simule une récupération réseau/calcule du défi du jour
        isLoading = true
        // Si settings déjà présents, on les garde, sinon on initialise à J-3 pour l'exemple
        val settings = currentSettings ?: DailyChallengeSettings(startDate = LocalDate.now().minusDays(3))
        currentSettings = settings
        currentChallenge = DailyChallenge(scheduledDate = LocalDate.now())
        isLoading = false
    }
}

// ---------- Routing ----------
sealed class DailyContentRoute {
    data class Intro(val showConnectButton: Boolean) : DailyContentRoute()
    data class Paywall(val day: Int) : DailyContentRoute()
    data object Main : DailyContentRoute()
    data class Error(val message: String) : DailyContentRoute()
    data object Loading : DailyContentRoute()
}

object DailyContentRouteCalculator {
    fun calculateRoute(
        hasConnectedPartner: Boolean,
        hasSeenIntro: Boolean,
        shouldShowPaywall: Boolean,
        paywallDay: Int,
        serviceHasError: Boolean,
        serviceErrorMessage: String?,
        serviceIsLoading: Boolean
    ): DailyContentRoute {
        if (serviceHasError) return DailyContentRoute.Error(serviceErrorMessage ?: "Unknown error")
        if (serviceIsLoading) return DailyContentRoute.Loading
        if (!hasConnectedPartner) return DailyContentRoute.Intro(showConnectButton = true)
        if (!hasSeenIntro) return DailyContentRoute.Intro(showConnectButton = false)
        if (shouldShowPaywall) return DailyContentRoute.Paywall(paywallDay)
        return DailyContentRoute.Main
    }
}

// ---------- Helpers ----------
private fun calculateExpectedDay(settings: DailyChallengeSettings): Int {
    val today = LocalDate.now(settings.zoneId)
    val days = ChronoUnit.DAYS.between(settings.startDate, today).toInt()
    return max(1, days + 1)
}

private fun isToday(date: LocalDate): Boolean = date == LocalDate.now()

private fun logAnalyticsEvent(context: Context, name: String, params: Bundle.() -> Unit = {}) {
    val bundle = Bundle().apply(params)
    FirebaseAnalytics.getInstance(context).logEvent(name, bundle)
}

// ---------- Flow principal ----------
@Composable
fun DailyChallengeFlowScreen(
    appState: AppState,
    dailyChallengeService: DailyChallengeService = remember { DailyChallengeService() }
) {
    val context = LocalContext.current

    // Configuration "onAppear"
    LaunchedEffect(appState, dailyChallengeService.currentChallenge, dailyChallengeService.currentSettings) {
        configureServiceIfNeeded(appState, dailyChallengeService)
    }

    // État dérivé pour la route actuelle
    val currentRoute by remember(
        appState,
        dailyChallengeService.isLoading,
        dailyChallengeService.currentChallenge,
        dailyChallengeService.currentSettings
    ) {
        mutableStateOf(
            run {
                val hasConnectedPartner = appState.currentUser?.partnerId?.trim()?.isNotEmpty() == true
                val currentDay = dailyChallengeService.currentSettings?.let { calculateExpectedDay(it) } ?: 1
                val isSubscribed = appState.currentUser?.isSubscribed == true
                val shouldShowPaywall =
                    if (isSubscribed) false else appState.freemiumManager?.canAccessDailyChallenge(currentDay)?.not() ?: false
                val serviceIsLoading = dailyChallengeService.isLoading && dailyChallengeService.currentChallenge == null

                DailyContentRouteCalculator.calculateRoute(
                    hasConnectedPartner = hasConnectedPartner,
                    hasSeenIntro = appState.introFlags.dailyChallenge,
                    shouldShowPaywall = shouldShowPaywall,
                    paywallDay = currentDay,
                    serviceHasError = false,
                    serviceErrorMessage = null,
                    serviceIsLoading = serviceIsLoading
                )
            }
        )
    }

    when (val route = currentRoute) {
        is DailyContentRoute.Intro -> {
            DailyChallengeIntroView(
                showConnectButton = route.showConnectButton,
                onShown = {
                    if (!route.showConnectButton) {
                        logAnalyticsEvent(context, "intro_shown") {
                            putString("screen", "daily_challenge")
                        }
                    }
                }
            )
        }
        is DailyContentRoute.Paywall -> {
            DailyChallengePaywallView(
                appState = appState,
                challengeDay = route.day
            )
        }
        is DailyContentRoute.Error -> {
            DailyChallengeErrorView(
                message = route.message,
                onRetry = { configureServiceIfNeeded(appState, dailyChallengeService) }
            )
        }
        DailyContentRoute.Loading -> DailyChallengeLoadingView()
        DailyContentRoute.Main -> DailyChallengeMainView()
    }
}

// Reprise fidèle de la logique Swift configureServiceIfNeeded()
private fun configureServiceIfNeeded(
    appState: AppState,
    dailyChallengeService: DailyChallengeService
) {
    val currentUser = appState.currentUser
    val partnerId = currentUser?.partnerId?.trim()

    // Partenaire requis
    if (partnerId.isNullOrEmpty()) {
        // En attente connexion partenaire
        return
    }
    // L'intro doit avoir été vue
    if (!appState.introFlags.dailyChallenge) {
        // En attente intro utilisateur
        return
    }

    // Si on a déjà un défi d'aujourd'hui, stop
    dailyChallengeService.currentChallenge?.let { challenge ->
        if (isToday(challenge.scheduledDate)) {
            return
        }
    }

    // Calcul du jour courant
    val currentDay = dailyChallengeService.currentSettings?.let { calculateExpectedDay(it) } ?: 1

    // Gérer le freemium AVANT de configurer le service
    appState.freemiumManager?.handleDailyChallengeAccess(currentDay) {
        dailyChallengeService.configure(appState)
    } ?: run {
        // Si pas de freemium manager, on configure directement
        dailyChallengeService.configure(appState)
    }
}

// ---------- Composants UI ----------

@Composable
private fun DailyChallengeIntroView(
    showConnectButton: Boolean,
    onShown: () -> Unit
) {
    LaunchedEffect(Unit) { onShown() }

    Box(
        modifier = Modifier
            .fillMaxSize()
            .background(PaywallBackdrop),
        contentAlignment = Alignment.Center
    ) {
        Column(
            horizontalAlignment = Alignment.CenterHorizontally,
            modifier = Modifier.padding(24.dp)
        ) {
            Text(
                text = "Daily Challenge",
                style = MaterialTheme.typography.headlineMedium.copy(fontWeight = FontWeight.Bold),
                color = Color.Black,
                textAlign = TextAlign.Center
            )
            Spacer(Modifier.height(12.dp))
            Text(
                text = if (showConnectButton)
                    "Connecte ton/ta partenaire pour commencer."
                else
                    "Bienvenue ! Découvre un défi par jour.",
                style = MaterialTheme.typography.bodyLarge,
                color = Color.Black.copy(alpha = 0.7f),
                textAlign = TextAlign.Center
            )
            if (showConnectButton) {
                Spacer(Modifier.height(24.dp))
                Button(
                    onClick = { /* TODO: navigation vers l’écran de connexion partenaire */ },
                    shape = RoundedCornerShape(28.dp)
                ) {
                    Text("Connecter mon/ma partenaire")
                }
            }
        }
    }
}

@Composable
private fun DailyChallengeMainView() {
    Box(
        modifier = Modifier
            .fillMaxSize()
            .background(PaywallBackdrop),
        contentAlignment = Alignment.Center
    ) {
        Text(
            text = "Daily Challenge — Main",
            style = MaterialTheme.typography.headlineSmall.copy(fontWeight = FontWeight.SemiBold)
        )
    }
}

@Composable
private fun DailyChallengeErrorView(
    message: String,
    onRetry: () -> Unit
) {
    Box(
        modifier = Modifier
            .fillMaxSize()
            .background(PaywallBackdrop),
        contentAlignment = Alignment.Center
    ) {
        Column(horizontalAlignment = Alignment.CenterHorizontally) {
            Text(
                text = "Erreur",
                style = MaterialTheme.typography.headlineSmall.copy(fontWeight = FontWeight.Bold),
                color = Color.Black
            )
            Spacer(Modifier.height(8.dp))
            Text(
                text = message,
                style = MaterialTheme.typography.bodyMedium,
                color = Color.Black.copy(alpha = 0.7f),
                textAlign = TextAlign.Center,
                modifier = Modifier.padding(horizontal = 24.dp)
            )
            Spacer(Modifier.height(16.dp))
            OutlinedButton(onClick = onRetry, shape = RoundedCornerShape(24.dp)) {
                Text("Réessayer")
            }
        }
    }
}

@Composable
private fun DailyChallengeLoadingView() {
    Box(
        modifier = Modifier
            .fillMaxSize()
            .background(PaywallBackdrop),
        contentAlignment = Alignment.Center
    ) {
        CircularProgressIndicator()
    }
}

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun DailyChallengePaywallView(
    appState: AppState,
    challengeDay: Int
) {
    val context = LocalContext.current
    var showSubscriptionSheet by remember { mutableStateOf(false) }

    Scaffold(
        containerColor = PaywallBackdrop,
        topBar = {
            Box(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(top = 40.dp, bottom = 24.dp),
                contentAlignment = Alignment.Center
            ) {
                Text(
                    text = stringResource(id = R.string.paywall_page_title_challenges),
                    style = MaterialTheme.typography.headlineMedium.copy(fontWeight = FontWeight.Bold),
                    color = Color.Black
                )
            }
        }
    ) { innerPadding ->
        Column(
            modifier = Modifier
                .padding(innerPadding)
                .fillMaxSize()
                .verticalScroll(rememberScrollState()),
            horizontalAlignment = Alignment.CenterHorizontally
        ) {

            Spacer(Modifier.height(60.dp))

            Text(
                text = stringResource(id = R.string.paywall_challenges_title),
                style = MaterialTheme.typography.titleLarge.copy(fontWeight = FontWeight.Bold),
                color = Color.Black,
                textAlign = TextAlign.Center,
                modifier = Modifier.padding(horizontal = 20.dp)
            )

            Spacer(Modifier.height(30.dp))

            // --------- Carte défi (floutée) ----------
            Box(
                modifier = Modifier
                    .padding(horizontal = 20.dp)
                    .fillMaxWidth()
                    .heightIn(min = 260.dp)
                    .clip(RoundedCornerShape(20.dp))
                    .shadow(10.dp, RoundedCornerShape(20.dp))
            ) {
                // Contenu "réel" de la carte
                Column(
                    modifier = Modifier
                        .fillMaxSize()
                        .background(CardBodyGradient)
                ) {
                    // Header
                    Box(
                        modifier = Modifier
                            .fillMaxWidth()
                            .background(CardHeaderGradient)
                            .padding(vertical = 20.dp),
                        contentAlignment = Alignment.Center
                    ) {
                        Text(
                            text = stringResource(id = R.string.daily_challenges_title),
                            style = MaterialTheme.typography.titleMedium.copy(
                                fontWeight = FontWeight.Bold,
                                color = Color.White
                            ),
                            textAlign = TextAlign.Center
                        )
                    }

                    // Corps
                    Column(
                        modifier = Modifier
                            .fillMaxWidth()
                            .weight(1f),
                        horizontalAlignment = Alignment.CenterHorizontally,
                        verticalArrangement = Arrangement.SpaceBetween
                    ) {
                        Spacer(Modifier.height(20.dp))
                        Text(
                            text = "Envoyez-lui un message pour lui dire pourquoi vous êtes reconnaissant de l'avoir dans votre vie aujourd'hui et partagez trois choses spécifiques que vous appréciez chez lui.",
                            style = MaterialTheme.typography.titleMedium.copy(
                                fontWeight = FontWeight.Medium,
                                color = Color.White
                            ),
                            textAlign = TextAlign.Center,
                            modifier = Modifier
                                .padding(horizontal = 30.dp)
                                .blur(8.dp) // FLOU sur le texte
                        )
                        Spacer(Modifier.height(12.dp))
                        Text(
                            text = "🎯  " + (
                                // Si tu ajoutes `day_number` = "Jour %1$d" dans strings.xml
                                // stringResource(id = R.string.day_number, challengeDay)
                                "Jour $challengeDay"
                            ),
                            style = MaterialTheme.typography.bodyLarge.copy(
                                fontWeight = FontWeight.SemiBold,
                                color = Color.White.copy(alpha = 0.9f)
                            ),
                            modifier = Modifier
                                .padding(bottom = 30.dp)
                                .blur(6.dp) // FLOU sur le footer
                        )
                    }
                }

                // Overlay "glassmorphism + blur" sur le corps uniquement
                Column(modifier = Modifier.fillMaxSize()) {
                    // Laisse l'entête (≈ 60dp) lisible
                    Spacer(Modifier.height(60.dp))
                    Box(
                        modifier = Modifier
                            .fillMaxSize()
                            .background(
                                Brush.verticalGradient(
                                    listOf(
                                        Color(0xFF33242B).copy(alpha = 0.95f),
                                        Color(0xFF664055).copy(alpha = 0.95f),
                                        Color(0xFF99523D).copy(alpha = 0.95f)
                                    )
                                )
                            )
                            .blur(15.dp) // vrai effet de flou
                            .background(Color.White.copy(alpha = 0.35f))
                            .clip(RoundedCornerShape(20.dp)),
                        contentAlignment = Alignment.Center
                    ) {
                        Text(
                            text = "💕",
                            style = MaterialTheme.typography.displaySmall.copy(color = Color.White),
                            modifier = Modifier
                        )
                    }
                }
            }

            Spacer(Modifier.height(20.dp))

            Text(
                text = stringResource(id = R.string.paywall_challenges_subtitle),
                style = MaterialTheme.typography.bodyMedium,
                color = Color.Black.copy(alpha = 0.7f),
                textAlign = TextAlign.Center,
                modifier = Modifier
                    .padding(horizontal = 30.dp)
            )

            Spacer(Modifier.height(40.dp))

            Button(
                onClick = {
                    showSubscriptionSheet = true
                    logAnalyticsEvent(context, "cta_premium_clicked") {
                        putString("source", "daily_challenge_paywall")
                        putInt("challenge_day", challengeDay)
                    }
                },
                shape = RoundedCornerShape(28.dp),
                modifier = Modifier
                    .height(56.dp)
                    .padding(horizontal = 24.dp),
                colors = ButtonDefaults.buttonColors(containerColor = PinkPrimary)
            ) {
                Text(
                    text = stringResource(id = R.string.paywall_continue_button),
                    style = MaterialTheme.typography.titleMedium.copy(fontWeight = FontWeight.SemiBold),
                    color = Color.White
                )
            }

            Spacer(Modifier.height(160.dp))
        }
    }

    if (showSubscriptionSheet) {
        ModalBottomSheet(onDismissRequest = { showSubscriptionSheet = false }) {
            SubscriptionSheetContent(onClose = { showSubscriptionSheet = false })
        }
    }
}

@Composable
private fun SubscriptionSheetContent(onClose: () -> Unit) {
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .padding(20.dp),
        horizontalAlignment = Alignment.CenterHorizontally
    ) {
        Text(
            text = "Premium",
            style = MaterialTheme.typography.titleLarge.copy(fontWeight = FontWeight.Bold)
        )
        Spacer(Modifier.height(8.dp))
        Text(
            text = "Débloque tous les défis quotidiens et plus encore.",
            style = MaterialTheme.typography.bodyMedium,
            textAlign = TextAlign.Center
        )
        Spacer(Modifier.height(16.dp))
        Button(onClick = onClose, shape = RoundedCornerShape(24.dp)) {
            Text("Fermer")
        }
        Spacer(Modifier.height(12.dp))
    }
}
